executor/plan_recreator.go

```diff
@@ -113,28 +153,20 @@ func (e *PlanRecreatorSingleExec) Open(ctx context.Context) error {
 	return nil
 }
 
-// Process dose the export/import work for reproducing sql queries.
-func (e *PlanRecreatorSingleInfo) Process() (string, error) {
-	// TODO: plan recreator load will be developed later
-	if e.Load {
-		return "", nil
-	}
-	return e.dumpSingle()
-}
-
 func (e *PlanRecreatorSingleInfo) dumpSingle() (string, error) {
 	// Create path
-	err := os.MkdirAll(recreatorPath, os.ModePerm)
+	path := fmt.Sprintf("%s/%v", recreatorPath, e.Ctx.GetSessionVars().ConnectionID)
+	err := os.MkdirAll(path, os.ModePerm)
 	if err != nil {
-		return "", errors.New("plan Recreator: cannot create plan recreator path")
+		return "", errors.New("Plan Recreator: cannot create plan recreator path")
 	}
 
 	// Create zip file
 	startTime := time.Now()
 	fileName := fmt.Sprintf("recreator_single_%v.zip", startTime.UnixNano())
-	zf, err := os.Create(recreatorPath + "/" + fileName)
+	zf, err := os.Create(path + "/" + fileName)
```

Comment:
Please use `filepath.Join(path, fileName)` here, which is safer than `path + "/" + fileName`.